tests/system.rs

```diff
@@ -431,3 +433,33 @@ fn rustfmt_diff_no_diff_test() {
     let diff = make_diff("a\nb\nc\nd", "a\nb\nc\nd", 3);
     assert_eq!(diff, vec![]);
 }
+
+// Compare strings without distinguishing between CRLF and LF
+fn string_eq_ignore_newline_repr(left: &str, right: &str) -> bool {
+    let left = BytesIgnoreNewlineRepr(left.bytes().peekable());
```

Comment:
I think it would be better to use `chars` rather than `bytes` here, and iterate over `char`s (I'm not sure if it is possible, but I feel like, there might be some subtle bug if `b'\r'` is a byte in a longer char